src/winrpath.cxx

```diff
@@ -0,0 +1,479 @@
+    if(!hMapObject){
+        std::cerr << "Unable to create mapping object\n";
+        return -5;
+    }
+    LPVOID basepointer = (char*)MapViewOfFile(hMapObject, FILE_MAP_WRITE, 0, 0, 0);
+    if(!basepointer){
+        std::cerr << "Unable to create file map view\n";
+        return -6;
+    }
+    // Establish base PE headers
+    PIMAGE_DOS_HEADER dos_header = (PIMAGE_DOS_HEADER)basepointer;
+    PIMAGE_NT_HEADERS nt_header = 
+        (PIMAGE_NT_HEADERS)((DWORD)basepointer + dos_header->e_lfanew);
+
+    PIMAGE_FILE_HEADER coff_header = 
+        (PIMAGE_FILE_HEADER)((DWORD)basepointer + dos_header->e_lfanew + sizeof(nt_header->Signature));
+        
+    PIMAGE_OPTIONAL_HEADER optional_header = 
+        (PIMAGE_OPTIONAL_HEADER)((DWORD)basepointer + dos_header->e_lfanew + sizeof(nt_header->Signature) + sizeof(nt_header->FileHeader));
+        
+    PIMAGE_SECTION_HEADER section_header = 
+        (PIMAGE_SECTION_HEADER)((DWORD)basepointer + dos_header->e_lfanew + sizeof(nt_header->Signature) + sizeof(nt_header->FileHeader) + sizeof(nt_header->OptionalHeader));
+    
+    DWORD number_of_rva_and_sections = optional_header->NumberOfRvaAndSizes;
+    if(number_of_rva_and_sections == 0) {
+        std::cerr << "PE file does not import symbols" << "\n";
+        return -1;
+    }
+    else if(number_of_rva_and_sections < 2) {
+        std::cerr << "PE file contains insufficient data directories, likely corrupted" << "\n";
+        return -2;
+    }
+
+    DWORD number_of_sections = coff_header->NumberOfSections;
+    // Data directory #2 points to the RVA of the import section
+    DWORD RVA_import_directory = nt_header->OptionalHeader.DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].VirtualAddress;
+    DWORD import_section_file_offset = RvaToFileOffset(section_header, number_of_sections, RVA_import_directory);
+    DWORD import_table_offset = (DWORD)basepointer + import_section_file_offset;
+    PIMAGE_IMPORT_DESCRIPTOR import_image_descriptor = (PIMAGE_IMPORT_DESCRIPTOR)(import_table_offset);
+    //DLL Imports
+    for (; import_image_descriptor->Name != 0; import_image_descriptor++) {
+        DWORD Imported_DLL = import_table_offset + (import_image_descriptor->Name - RVA_import_directory);
+        std::ostringstream str_stream;
+        str_stream << Imported_DLL;
+        if(this->spack_check_for_dll(str_stream.str())) {
+            if(!this->rename_dll(Imported_DLL, str_stream.str())) {
+                throw SpackException("Unable to relocate DLL");
+            }
+        }
+    }
+    if(!CloseHandle(basepointer)){
+        std::cerr << "Unable to properly close file view\n";
+        return -3;
+    }
+    if(!CloseHandle(hMapObject)){
+        std::cerr << "Unable to properly close file map\n";
+        return -4;
+    }
+}
+
+LibRename::LibRename(std::string lib, bool full, bool deploy, bool replace)
+: replace(replace), full(full), lib(lib), deploy(deploy)
+{
```

Comment:
I see `full`/`deploy` have effects, but it would be useful for me (now, while I'm reviewing it) to know what their intended purpose is (in docstring form).

Reply:
Will document!